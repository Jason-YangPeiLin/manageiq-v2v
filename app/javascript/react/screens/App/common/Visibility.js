import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

// This component has no role on the page, and will be hidden from screen readers (role="presentation").
// Renders the children wrapped in visibility: hidden if the condition is met.
// Either way, the children take up space, visibility will not change the flow of the layout.
// The onKeyUp handler and role are used to satisfy eslint-plugin-jsx-a11y.
const Visibility = ({ children, hidden }) => (
  <span style={{ visibility: hidden ? 'hidden' : '' }} onKeyUp={noop} role="presentation">
    {children}
  </span>
);

Visibility.propTypes = {
  children: PropTypes.node,
  hidden: PropTypes.bool
};

export default Visibility;
