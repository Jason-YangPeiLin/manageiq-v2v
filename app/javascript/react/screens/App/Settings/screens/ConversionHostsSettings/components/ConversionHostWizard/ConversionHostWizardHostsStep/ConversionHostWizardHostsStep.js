import React from 'react';
import PropTypes from 'prop-types';
import { reduxForm, Field } from 'redux-form';
import { length } from 'redux-form-validators';
import { Form } from 'patternfly-react';
import { RHV, OPENSTACK } from '../../../../../../../../../common/constants';
import TypeAheadSelectField from '../../../../../../common/forms/TypeAheadSelectField';
import { stepIDs } from '../ConversionHostWizardConstants';

const ConversionHostWizardHostsStep = ({ selectedProviderType, selectedCluster }) => {
  let hostOptions = [];
  let emptyLabel = '';
  if (selectedProviderType === RHV) {
    hostOptions = selectedCluster.hosts;
    emptyLabel = __('No hosts available for the selected cluster.');
  }
  if (selectedProviderType === OPENSTACK) {
    hostOptions = selectedCluster.vms;
    emptyLabel = __('No hosts available for the selected project.');
  }
  return (
    <Form className="form-vertical">
      <Form.FormGroup controlId="host-selection">
        <Form.ControlLabel>{__('Hosts to configure as conversion hosts')}</Form.ControlLabel>
        <Field
          component={TypeAheadSelectField}
          name="hosts"
          controlId="host-selection"
          multiple
          clearButton
          options={hostOptions}
          labelKey="name"
          placeholder={__('Select one or more hosts...')}
          emptyLabel={hostOptions.length === 0 ? emptyLabel : __('No matches found.')}
          highlightOnlyResult
          selectHintOnEnter
          validate={[length({ min: 1 })]}
        />
      </Form.FormGroup>
    </Form>
  );
};

ConversionHostWizardHostsStep.propTypes = {
  selectedProviderType: PropTypes.string,
  selectedCluster: PropTypes.object
};

export default reduxForm({
  form: stepIDs.hostsStep,
  destroyOnUnmount: false,
  initialValues: { hosts: [] }
})(ConversionHostWizardHostsStep);
