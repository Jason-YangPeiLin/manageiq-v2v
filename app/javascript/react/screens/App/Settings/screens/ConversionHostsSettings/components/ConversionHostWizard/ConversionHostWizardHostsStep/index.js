import ConversionHostWizardHostsStep from './ConversionHostWizardHostsStep';

export default ConversionHostWizardHostsStep;
