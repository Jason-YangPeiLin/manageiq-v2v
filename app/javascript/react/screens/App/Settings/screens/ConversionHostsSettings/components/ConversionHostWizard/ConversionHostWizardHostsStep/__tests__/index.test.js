import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { reducer as formReducer } from 'redux-form';

import { generateStore } from '../../../../../../../common/testReduxHelpers';
import ConversionHostWizardHostsStep from '../ConversionHostWizardHostsStep';
import ConversionHostWizardHostsStepContainer from '../index';
import { RHV } from '../../../../../../../../../../common/constants';

describe('ConversionHostWizardHostsStep integration test', () => {
  const store = generateStore(
    { form: formReducer },
    {
      form: { conversionHostWizardLocationStep: { values: { providerType: RHV, cluster: '1' } } },
      targetResources: { targetClusters: [{ mock: 'cluster', id: '1', hosts: [] }] },
      settings: {
        conversionHosts: [{ mock: 'conversionHost1' }, { mock: 'conversionHost2' }],
        conversionHostTasksByResource: { mock: 'tasksByResource' }
      }
    }
  );

  const mountComponent = () =>
    mount(
      <Provider store={store}>
        <ConversionHostWizardHostsStepContainer />
      </Provider>
    );

  it('should mount ConversionHostWizardHostsStep with mapStateToProps reduced', () => {
    const wrapper = mountComponent();

    // query the unconnected component to assert reduced props
    const component = wrapper.find(ConversionHostWizardHostsStep);
    expect(component.props()).toMatchSnapshot();
  });
});
