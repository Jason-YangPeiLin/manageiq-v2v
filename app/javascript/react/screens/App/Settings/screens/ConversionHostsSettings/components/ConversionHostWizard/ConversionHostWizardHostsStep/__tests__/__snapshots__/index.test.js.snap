// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversionHostWizardHostsStep integration test should mount ConversionHostWizardHostsStep with mapStateToProps reduced 1`] = `
Object {
  "conversionHostTasksByResource": Object {
    "mock": "tasksByResource",
  },
  "conversionHosts": Array [
    Object {
      "mock": "conversionHost1",
    },
    Object {
      "mock": "conversionHost2",
    },
  ],
  "selectedCluster": Object {
    "hosts": Array [],
    "id": "1",
    "mock": "cluster",
  },
  "selectedProviderType": "rhevm",
}
`;
